### PR TITLE
ci(clang-tidy-differential): point to the new config

### PR DESCRIPTION
**Pull request by xmfcx**
## Description

This new config file we are pointing is located at`autoware/main/.clang-tidy-ci`.

It is supposed to be less restrictive to make it a required check.

And we will make it more strict over time as we adjust existing code to adhere these new standards.

## Related links

**Follow up from:**

- https://github.com/autowarefoundation/autoware/pull/4918

**Parent discussion:**
- [Proposal to enable clang-tidy as a required CI for Autoware #4876](https://github.com/orgs/autowarefoundation/discussions/4876)

## How was this PR tested?

- https://github.com/autowarefoundation/autoware.universe/actions/runs/9697046266?pr=7729

## Notes for reviewers

- https://github.com/autowarefoundation/autoware.universe/pull/7729/files#r1657097594

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by xmfcx** [comment]
@veqcc it failed the test:
- https://github.com/autowarefoundation/autoware.universe/actions/runs/9697046266/job/26761095747?pr=7729#step:9:447

I think you need to either:
- update the config file in `autoware/main/.clang-tidy-ci`
- or fix the packages to conform to the new rules
before we can make it a `Required` workflow.

**Comment by veqcc** [comment]
@xmfcx 

> @veqcc it failed the test:
> 
> * https://github.com/autowarefoundation/autoware.universe/actions/runs/9697046266/job/26761095747?pr=7729#step:9:447
> 
> I think you need to either:
> 
> * update the config file in `autoware/main/.clang-tidy-ci`
> * or fix the packages to conform to the new rules
>   before we can make it a `Required` workflow.

Thank you for informing me!
I have confirmed the workflow result. It seems the clang-tidy just warns and invokes no errors, which is the expected behavior.
I have to check why the `Run clang-tidy` step ends up with exit code 1. I'll do it tomorrow 👍

**Comment by xmfcx** [comment]
> I have to check why the Run clang-tidy step ends up with exit code 1. I'll do it tomorrow

- https://github.com/autowarefoundation/autoware-github-actions/blob/6c71a4d6718a29d385cfb22ce6ac94ad1ad8f543/clang-tidy/action.yaml#L129
   - `clang-tidy -p build/ -export-fixes /tmp/clang-tidy-result/fixes.yaml ${{ steps.get-target-files.outputs.target-files }} || true`
- https://github.com/autowarefoundation/autoware-github-actions/blob/6c71a4d6718a29d385cfb22ce6ac94ad1ad8f543/clang-tidy/action.yaml#L148-L152
```yaml
    - name: Mark the workflow as failed if the fixes.yaml file exists
      if: ${{ steps.check-fixes-yaml-existence.outputs.exists == 'true' }}
      run: |
        exit 1
      shell: bash
```

This is the reason it is making the job fail. I'll leave the rest to you.

**Comment by veqcc** [comment]
> This is the reason it is making the job fail. I'll leave the rest to you.

I have fixed in https://github.com/autowarefoundation/autoware-github-actions/pull/306

File: .github/workflows/build-and-test-differential.yaml [hunk 1]
```diff
@@ -154,8 +154,8 @@ jobs:
         with:
           rosdistro: humble
           target-packages: ${{ steps.get-modified-packages.outputs.modified-packages }}
-          target-files: ${{ steps.get-changed-files.outputs.changed-files }}
-          clang-tidy-config-url: https://raw.githubusercontent.com/autowarefoundation/autoware/main/.clang-tidy
+          target-files: ${{ steps.get-modified-files.outputs.changed_files }}
```
Comment (xmfcx) [inline .github/workflows/build-and-test-differential.yaml:157]:
```suggestion
          target-files: ${{ steps.get-changed-files.outputs.changed-files }}
```

File: sensing/gnss_poser/src/gnss_poser_core.cpp [hunk 2]
```diff
@@ -24,6 +24,8 @@
 #include <string>
 #include <vector>
 
+// TODO(mfc): remove this comment before merging this pr, this is just for testing
+
```
[the remark this answers is not in the document]
Comment (xmfcx) [reply sensing/gnss_poser/src/gnss_poser_core.cpp:28]:
```suggestion
```

**Pull request merged**